### PR TITLE
[move source language] Added numeric cast support

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Added support for numeric cast operation
- The types supported by the VM are: u8, u64, u128

## Test Plan

- Wrote tests

### CODE REVIEW

File: language/move-lang/src/parser/syntax.rs [hunk 1]
```diff
@@ -612,11 +612,7 @@ fn parse_term<'input>(tokens: &mut Lexer<'input>) -> Result<Exp, Error> {
                     Exp_::Annotate(Box::new(e), ty)
                 } else if match_token(tokens, Tok::As)? {
                     let ty = parse_type(tokens)?;
-                    // FIXME: Exp_::Cast(Box::new(e), ty)
-                    // The extra let binding here is just to keep clippy from complaining
-                    // that this code is the same as above. Remove it when there is a Cast
-                    // expression (or whatever we end up calling it).
-                    let e = Exp_::Annotate(Box::new(e), ty);
+                    let e = Exp_::Cast(Box::new(e), ty);
```
Comment (bob-wilson) [inline language/move-lang/src/parser/syntax.rs]: Can you get rid of the bogus let binding here? I was just trying to shut up clippy.

File: language/move-lang/src/hlir/translate.rs [hunk 2]
```diff
@@ -987,7 +987,29 @@ fn exp_impl(context: &mut Context, result: &mut Block, e: T::Exp) -> H::Exp {
...
             HE::BorrowLocal(mut_, tmp)
         }
-
+        TE::Cast(te, rhs_ty) => {
+            use H::BaseType_ as B;
+            use H::SingleType_ as S;
+            use H::Type_ as T;
+            use N::BuiltinTypeName_ as BT;
+            use N::TypeName_ as TN;
+            macro_rules! builtin {
```
Comment (bob-wilson) [inline language/move-lang/src/hlir/translate.rs]: This is pretty fancy (i.e., hard to follow). I haven't done anything with this hlir phase before, but I'm surprised that we don't already have some code to translate these builtin types. I guess this is fine for now, but you might consider splitting this into a separate function or adding some comments to explain what is going on here and why it is done this way.

Comment (tnowacki) [reply language/move-lang/src/hlir/translate.rs]: I'll make a function thats less confusing
